backend/bin/deal-observer-backend.js

```diff
@@ -108,20 +109,48 @@ const sparkApiSubmitDealsLoop = async (pgPool, { sparkApiBaseUrl, sparkApiToken,
 
     if (INFLUXDB_TOKEN) {
       recordTelemetry(`loop_${slug(LOOP_NAME, '_')}`, point => {
-        point.intField('interval_ms', SPARK_API_SUBMIT_DEALS_LOOP_INTERVAL)
+        point.intField('interval_ms', LOOP_INTERVAL)
         point.intField('duration_ms', dt)
       })
       recordTelemetry('submitted_deals_stats', point => {
         point.intField('submitted_deals', numberOfSubmittedDeals)
       })
     }
-    if (dt < OBSERVE_ACTOR_EVENTS_LOOP_INTERVAL) {
-      await timers.setTimeout(SPARK_API_SUBMIT_DEALS_LOOP_INTERVAL - dt)
+    if (dt < LOOP_INTERVAL) {
+      await timers.setTimeout(LOOP_INTERVAL - dt)
+    }
+  }
+}
+
+export const pieceIndexerLoop = async (makeRpcRequest, makePixRequest, pgPool) => {
+  const LOOP_NAME = 'Piece Indexer'
+  while (true) {
+    const start = Date.now()
+    const queryLimit = 1000
```

Comment:
How did we come up with this magic number? It defines how many active deals we look at at a time. It looks like by tuning it we control the trade off between performance and memory usage (but not concurrency).

Reply:
I don't remember the details; it's best to ask @NikolasHaimerl. Your description looks correct to me.

This will be easy to tweak later, I think we should pick some reasonable number for now and move on. WDYT?